File: openapi/cpp_client/examples/Secrets.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <map>

#include <Secrets.h>

Secrets::Secrets() {}

// Getter method to get a value corresponding to a given key from config.properties
std::string Secrets::get(const std::string& key) {
    std::ifstream input("./resources/config.properties");
    std::map<std::string, std::string> prop;

    if (!input) {
        throw std::runtime_error("Failed to open config.properties");
    }

    std::string line;
    while (std::getline(input, line)) {
        std::size_t found = line.find_first_of(" = ");
        if (found != std::string::npos) {
            std::string key = line.substr(0, found);
            std::string value = line.substr(found + 3);
            prop[key] = value;
        }
    }

    input.close();

    auto it = prop.find(key);
    if (it != prop.end()) {
        return it->second;
    } else {
        throw std::runtime_error("Key not found in config.properties");
    }
}

File: openapi/java_client/src/main/java/examples/Secrets.java
package Examples;

import java.io.FileInputStream;
import java.io.IOException;
import java.util.Properties;

public class Secrets {

    /* Constructor */ 
    public Secrets(){
        
    }

    /* Helper method to open config.properties file */
    private Properties openFile() throws IOException{
        String filePath = "resources/config.properties";

        Properties prop = new Properties();
        FileInputStream input = new FileInputStream(filePath);
        prop.load(input);
        input.close();
        return prop;
    }

    /* Getter method to get a the value corresponding to a given key from config.properties */
    public String get(String key ) throws IOException{
        Properties prop = openFile();
        return prop.getProperty(key);
    }

}
